samples/mpsl/timeslot/src/main.c

```diff
@@ -72,7 +75,23 @@ static mpsl_timeslot_signal_return_param_t *mpsl_timeslot_callback(
 	mpsl_timeslot_signal_return_param_t *p_ret_val = NULL;
 
 	switch (signal_type) {
+
 	case MPSL_TIMESLOT_SIGNAL_START:
+		/* No return action */
+		signal_callback_return_param.callback_action =
+			MPSL_TIMESLOT_SIGNAL_ACTION_NONE;
+		p_ret_val = &signal_callback_return_param;
+
+		/* Setup timer to trigger an interrupt (and thus the TIMER0
+		 * signal) before timeslot end.
+		 */
+		nrf_timer_mode_set(NRF_TIMER0, NRF_TIMER_MODE_TIMER);
```

Comment:
I am not sure anymore whether timer0 has value of 0 on start signal. In any case it may be changed.
Therefore I suggest to capture current timer0 value first and then set relative CC value